lib/index.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Bedrock Agreement Module.
+ *
+ * Copyright (c) 2016 Digital Bazaar, Inc. All rights reserved.
+ */
+/* jshint node:true */
+'use strict';
+
+var async = require('async');
+var bedrock = require('bedrock');
+var brPermission = require('bedrock-permission');
+var database = require('bedrock-mongodb');
+var eventLog = require('bedrock-event-log').log;
+var BedrockError = bedrock.util.BedrockError;
+var uuid = require('uuid').v4;
+
+// load config defaults
+require('./config');
+
+bedrock.events.on('bedrock.test.configure', function() {
+  require('./test.config');
+});
+
+// module permissions
+var PERMISSIONS = bedrock.config.permission.permissions;
+
+// module API
+var api = {};
+module.exports = api;
+
+var logger = bedrock.loggers.get('app');
+
+/**
+ * Add an agreement event
+ *
+ * @param actor the Identity performing the action.
+ * @param agreements a string or array of strings referencing agreements
+ * @param callback(err, result) called once the operation completes
+ */
+api.add = function(actor, agreements, callback) {
+  if(!actor || !('id' in actor)) {
+    return callback(new TypeError(
+      'actor must be an object with an "id" property.'));
+  }
+  if(typeof agreements === 'string') {
+    agreements = [agreements];
+  } else if(!Array.isArray(agreements)) {
+    return callback(new TypeError(
+      'agreements parameter must be a string or an array.'));
+  }
+  var event = {
+    type: 'AgreementAccept',
+    date: new Date().toJSON(),
+    resource: agreements,
+    actor: actor.id
+  };
+  async.auto({
+    checkPermission: function(callback) {
+      brPermission.checkPermission(
+        actor, PERMISSIONS.AGREEMENT_CREATE,
+        {resource: event, translate: 'actor'}, callback);
+    },
+    insert: ['checkPermission', function(callback) {
+      eventLog.add(event, callback);
+    }]
+  }, function(err, results) {
+    if(err) {
+      return callback(err);
+    }
+    callback(null, results.insert);
+  });
+};
+
+/**
+ * Get all agreements accepted by an actor
+ *
+ * @param actor the Identity performing the action.
+ * @param callback(err, result) called with an error or an array of agreements
+ */
+api.get = function(actor, callback) {
```

Comment:
We might want this to be `getAccepted` or something. Otherwise people might confuse this with getting the text of an agreement. I think we can keep `AGREEMENT_ACCESS` as the permission. If we need more permissions in the future we can add and tweak.